project/plugins/bitbucket/__init__.py

```diff
@@ -15,56 +14,148 @@ def getSecret(configMap):
+
+             log = errorCheck(invGroup, plugin_tag, userName)
+
+             if log:
+                print(log)
+                instruction = log
+             else:
+                log = 'BitBucket: Email invite sent from Bitbucket.\n'
+                instruction = inviteMessage(configMap,plugin_tag)
+                done = True
+
+     return getJsonResponse('Bitbucket', email, log, instruction, done)
 
 def removeUser(email,configMap,allPermissions, plugin_tag):
-     #Get Authorization token
-     key=getKey(configMap)
+
+     done = False
+     userName = email.split('@', 1)[0]
      data = {'grant_type': 'client_credentials'}
-     credential = requests.post("https://bitbucket.org/site/oauth2/access_token", auth=(getKey(configMap),getSecret(configMap)), data=data)
-     my_json = credential.content.decode('utf8')
-     data = json.loads(my_json)
-     access_token=data.get('access_token')
-
-     #get all groups
-     groups=requests.get("https://api.bitbucket.org/1.0/groups/"+configMap['global']['organization']+"?access_token="+access_token)
-     my_json=groups.content.decode('utf8')
-     data = json.loads(my_json)
-
-     # Remove from groups
-     for group in data:
-         delMem= requests.delete("https://api.bitbucket.org/1.0/groups/"+configMap['global']['organization']+"/"+group.get('name').lower()+"/members/"+email+"?access_token="+access_token)
-
-     log = 'BitBucket: '+email+' removed from team.\n'
-     instruction = email.split('@', 1)[0] + removalMessage(configMap,plugin_tag)
-     return getJsonResponse('Bitbucket', email, log, instruction)
+     cli_groups = getGroups(configMap, plugin_tag)
+
+     credential = requestCredentials(configMap, data)
+     log = errorCheck(credential, plugin_tag, userName)
+
+     if log:
+         print(log)
+         instruction = log
+     else:
+         my_json = credential.content.decode('utf8')
+         data = json.loads(my_json)
+         access_token=data.get('access_token')
+
+         #get all groups
+         groups=requests.get("https://api.bitbucket.org/1.0/groups/"+configMap['global']['organization']+"?access_token="+access_token)
+         log = errorCheck(credential, plugin_tag, userName)
+
+         if log:
+             print(log)
+             instruction = log
+         else:
+             my_json=groups.content.decode('utf8')
+             data = json.loads(my_json)
+
+         if userExists(data, userName, cli_groups) == False:
+             log = ("The user " + userName + " does not exist, delete failed")
+             instruction = log
+             print(log)
+
+         else:
+             # Remove from groups
+             for group in data:
+                delMem= requests.delete("https://api.bitbucket.org/1.0/groups/"+configMap['global']['organization']+
+                                        "/"+group.get('name').lower()+"/members/"+email+"?access_token="+access_token)
+             log = errorCheck(delMem, plugin_tag, userName)
```

Comment:
Does this handle the case where a user may be in multiple groups?

Reply:
I believe it does